File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        min_num=prices[0]
        max_num=0

        for i in range(1, len(prices)):
            if prices[i] < min_num:
                min_num=prices[i]
            
            profit=prices[i] - min_num

            if profit>max_num:
                max_num=profit

        return max_num

        
        

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int profit=0;
        int buy=Integer.MAX_VALUE;

        for(int i=0;i<prices.length;i++){
            if(buy>prices[i]){
                buy=prices[i];
            }
            if(profit<(prices[i]-buy)){
                profit=prices[i]-buy;
            }
        }
        return profit;
    }
}